src/components/Map/index.js

```diff
@@ -77,11 +114,43 @@ const MapView = ({coordinates = null, zoomLevel = DEFAULT_ZOOM}) => {
     if (!coordinates || coordinates.length !== 2) return;
     if (!coordinates.filter(value => isFinite(value)).length === 2) return;
     map.setView(coordinates, zoomLevel);
-  });
+    // eslint-disable-next-line
+  }, [coordinates]);
   // rendering is done by leaflet, so just return null
   return null;
 };
 
+const SearchControl = props => {
```

Comment:
we prefer spreading the props already in the component definition:

```suggestion
const SearchControl = ({ onMarkerSet, ... }) => {
```
since this component is wrapping the `GeoSearchControl` object, I'd package those options as a separate prop instead of polluting the component prop namespace.

So do this instead:

```jsx
<SearchControl onMarkerSet={onMarkerSet} options={{'showMarker': false, ...}} />
```
which makes it easier to pass everything down.

Additionally, I see that you pass along the provider instance, but can't you just create that instance inside of this component? That way, everything related to the geo search is self-contained and you have less props to pass along.